### ISSUE TITLE
[Bug] [ray/tune] Best trail printing `None` when input a nested config

### ISSUE THREAD

**Issue by Eleven1Liu** [ray-project/ray#20901]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

1. **the bug**: When passing in the nested config (Ex. `config['network_config']['dropout']`) to `tune.run()`, parameters specified in the nested config are `None` in "Current best trial". 
Please see the screenshot in "useful information".

2.  **expected behavior**
It seems that when printing the best trial, it should `unflattened_lookup`  the `parameter_columns` like `trial_progress_table` does. For example,  `['network_config/dropout']` should be unflattened `['network_config']['dropout']`.

3. **useful information**
- screenshot
![image](https://user-images.githubusercontent.com/55077129/144812357-785049d0-f77f-4a9c-859c-da4c692d7cb5.png)

- code
Function `best_trail_str` is looking for `config['network_config/dropout']` (i.e., `p`) that doesn't exist. It should be unflattened to `config['network_config']['dropout']`. 
https://github.com/ray-project/ray/blob/b3a9d4d87d31255e0871852efaf1ddff7acdc390/python/ray/tune/progress_reporter.py#L790

### Versions / Dependencies

ray: 1.9.0
python: 3.7
OS: Linux

### Reproduction script

```python=
config = {
    "seed": 1337
    "epochs": 200
    "batch_size": 16
    "learning_rate": tune.grid_search([0.001, 0.003, 0.0001, 0.0003]),
    "network_config": {
         "num_filter_per_size": tune.grid_search([50, 150, 250, 350, 450, 550]),
         "filter_sizes": tune.grid_search([[2], [4], [6], [8], [10]]),
         "dropout":  tune.grid_search([0.2, 0.4, 0.6, 0.8])
    }
}

# Set up CLIReporter.
parameter_columns = {
    "learning_rate": "learning_rate", 
    "network_config/num_filter_per_size": "num_filter_per_size",
    "network_config/dropout": "dropout"
}
reporter = tune.CLIReporter(metric_columns=['val_P@1'],
                            parameter_columns=parameter_columns,
                            metric='val_P@1',
                            mode='max',
                            sort_by_metric=True)
# Run ray tune.
tune.run(
        tune.with_parameters(
            trainable_function,
            parameter_columns=parameter_columns,
            **data),
        ...
        progress_reporter=reporter,
        config=config)
```

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by Eleven1Liu** [comment]
Solved by #21078.
